src/auth/downscopedclient.ts

```diff
@@ -74,35 +87,44 @@ interface AvailabilityCondition {
   description?: string;
 }
 
+/**
+ * Downscoped with Credential Access Boundary Client.
+ * This is used to instantiate AuthClients for exchanging downscoped access
+ * tokens using GCP access token fetched from source credentials such as service
+ * account, user account or external account.
+ * The process of exchanging GCP access token for downscoped token is based
+ * on STS token endpoint.
+ */
 export class DownscopedClient extends AuthClient {
-  /**
-   * OAuth scopes for the GCP access token to use. When not provided,
-   * the default https://www.googleapis.com/auth/cloud-platform is
-   * used.
-   */
   private cachedDownscopedAccessToken: CredentialsWithResponse | null;
   private readonly stsCredential: sts.StsCredentials;
-  public readonly authClient: AuthClient;
-  public readonly credentialAccessBoundary: CredentialAccessBoundary;
   public readonly eagerRefreshThresholdMillis: number;
   public readonly forceRefreshOnFailure: boolean;
 
   constructor(
-    private client: AuthClient,
-    private cab: CredentialAccessBoundary,
+    private authClient: AuthClient,
+    private credentialAccessBoundary: CredentialAccessBoundary,
     additionalOptions?: RefreshOptions
   ) {
     super();
 
-    // Check a number of 1-10 access boundary rules are defined within credential access boundary.
-    if (cab.accessBoundary.accessBoundaryRules.length === 0) {
+    // Check 1-10 Access Boundary Rules are defined within Credential Access Boundary.
+    if (
+      credentialAccessBoundary.accessBoundary.accessBoundaryRules.length === 0
+    ) {
       throw new Error('At least one access boundary rule needs to be defined.');
-    } else if (cab.accessBoundary.accessBoundaryRules.length > 10) {
-      throw new Error('Access boundary rule exceeds limit, max 10 allowed.');
+    } else if (
+      credentialAccessBoundary.accessBoundary.accessBoundaryRules.length >
+      MAX_BOUNDARY_LIMIT
+    ) {
+      throw new Error(
+        `Access boundary rule exceeds limit, max ${MAX_BOUNDARY_LIMIT} allowed.`
```

Comment:
nit:  "The provided access boundary has more than ${MAX_BOUNDARY_LIMIT} access boundary rules."